verification/curator-service/ui/src/components/Users.tsx

```diff
@@ -85,113 +66,145 @@ class Users extends React.Component<Props, UsersState> {
+                            type: 'string',
+                            render: (rowData): JSX.Element =>
+                                this.rolesFormControl(rowData),
+                        },
+                    ]}
+                    data={(query): Promise<QueryResult<TableRow>> =>
+                        new Promise((resolve, reject) => {
+                            let listUrl = this.state.url;
+                            listUrl += '?limit=' + query.pageSize;
+                            listUrl += '&page=' + (query.page + 1);
+                            this.setState({ error: '' });
+                            const response = axios.get<ListResponse>(listUrl);
+                            response
+                                .then((result) => {
+                                    const flattenedUsers: TableRow[] = [];
+                                    const users = result.data.users;
+                                    this.setState({ users: users });
+                                    for (const c of users) {
+                                        flattenedUsers.push({
+                                            id: c._id,
+                                            name: c.name || 'Name not provided',
+                                            email: c.email,
+                                            roles: c.roles,
+                                        });
+                                    }
+                                    resolve({
+                                        data: flattenedUsers,
+                                        page: query.page,
+                                        totalCount: result.data.total,
+                                    });
+                                })
+                                .catch((e) => {
+                                    this.setState({ error: JSON.stringify(e) });
+                                    reject(e);
+                                });
+                        })
+                    }
+                    title="Users"
+                    options={{
+                        search: false,
+                        filtering: false,
+                        sorting: false,
+                        padding: 'dense',
+                        draggable: false, // No need to be able to drag and drop headers.
+                        pageSize: 10,
+                        pageSizeOptions: [5, 10, 20, 50, 100],
+                    }}
+                />
+            </Paper>
+        );
+    }
+
     updateRoles(
         event: React.ChangeEvent<{ value: string[] }>,
-        updatedUser: User,
+        userId: string,
     ): void {
         axios
-            .put(this.state.url + updatedUser._id, {
+            .put(this.state.url + userId, {
                 roles: event.target.value,
             })
             .then(() => {
                 const updatedUsers = this.state.users.slice();
```

Comment:
Now that it's a table can we get rid of lines 159 -> 166 (RHS) and let `this.tableRef.current.onQueryChange();` do the updating?

Reply:
you are right, much simpler now, thanks